src/preamble.js

```diff
@@ -2187,6 +2187,7 @@ function integrateWasmJS() {
 #else
       removeRunDependency('wasm-instantiate');
 #endif
+      stackRestore(STACKTOP);
```

Comment:
lgtm but please add a comment here, something like `set up the initial position of the stack`.

Reply:
Also maybe add `TODO: it is not necessary to pass in STACKTOP as an import when this is used`. Which is in the wasm backend case, so maybe this should be in `#ifdef WASM_BACKEND`.

Reply:
Seconding `#if WASM_BACKEND`